etcdadm/etcdadm

```diff
@@ -0,0 +1,1559 @@
+#!/bin/bash
+
+# To prevent etcdctl errors like "Error:  net/http: TLS handshake timeout" when ETCD_ENDPOINT is set in e.g. /etc/environment
+# Gotcha: ETCDCTL_ENDPOINT seems to be preferred over --peers flag, which results in `etcdctl --peers` to be useless
+unset ETCDCTL_ENDPOINT
+
+export | grep ETCDCTL
+
+ETCD_WORK_DIR=${ETCD_WORK_DIR:-$(pwd)/work}
+
+if [ "${DEBUG:-}" == "yes" ]; then
+  set -vx
+fi
+
+_info() {
+  echo "$0: info: ${FUNCNAME[1]}: $*" >&2
+}
+
+_error() {
+  echo "$0: error: ${FUNCNAME[1]}: $*" >&2
+}
+
+_panic() {
+  echo "$0: panic: ${FUNCNAME[1]}: $*" >&2
+  exit 1
+}
+
+_array_join() {
+  local r=""
+  for t in "${@}"; do
+    if [ "$r" != "" ]; then
+      r="$r \"$t\""
+    else
+      r="\"$t\""
+    fi
+  done
+  echo "$r"
+}
+
+_current_time() {
+  date +%s
+}
+
+_run_as_root() {
+  local status
+  if [[ "$EUID" > 0 ]]; then
+    echo running command as root: "$(_array_join "$@")"
+    _array_join "${@}" | sudo bash
+  else
+    _array_join "${@}" | bash
+  fi
+  status=$?
+  return "$status"
+}
+
+config_awscli_docker_image() {
+  echo "${ETCDADM_AWSCLI_DOCKER_IMAGE:-quay.io/coreos/awscli}"
+}
+
+awscli_rkt_image() {
+  echo "docker://$(config_awscli_docker_image)"
+}
+
+config_aws_region() {
```

Comment:
I guess these functions especially had been useful only when I had not set `set -e` :)
I'm ok with making them values as long as it can emit an undefined-variable-error where necessary.
I'll try later 👍